google/resource_compute_url_map.go

```diff
@@ -16,674 +33,796 @@ func resourceComputeUrlMap() *schema.Resource {
 		Delete: resourceComputeUrlMapDelete,
 
 		Importer: &schema.ResourceImporter{
-			State: resourceComputeUrlMapImportState,
+			State: resourceComputeUrlMapImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(240 * time.Second),
+			Update: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
 		},
 
 		Schema: map[string]*schema.Schema{
-			"default_service": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
+			"default_service": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
-
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-
-			"description": &schema.Schema{
+			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
-
-			"fingerprint": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-			},
-
-			"host_rule": &schema.Schema{
+			"host_rule": {
 				Type:     schema.TypeSet,
 				Optional: true,
-				// TODO(evandbrown): Enable when lists support validation
-				//ValidateFunc: validateHostRules,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"description": &schema.Schema{
-							Type:     schema.TypeString,
-							Optional: true,
-						},
-
-						"hosts": &schema.Schema{
+						"hosts": {
 							Type:     schema.TypeList,
```

Comment:
Hmm do you think it would make more sense if `hosts` and `paths` were sets?

Reply:
Done.